api/envoy/config/core/v3/address.proto

```diff
@@ -142,7 +147,7 @@ message BindConfig {
   // than one additional address specified, only the first address matched IP version will be
   // returned. If there is no same IP version address found, the address in the `source_address`
   // will be returned.
-  repeated SocketAddress additional_source_addresses = 4;
+  repeated AdditionalSourceAddress additional_source_addresses = 4;
```

Comment:
Maybe we can just use `repeated BindConfig` here?

Reply:
emm.. we still need to deprecate `additional_source_addresses`. If we put the additional addresses inside the BindConfig, we still have a place to add config which applies to all addresses.

Reply:
Two options in my mind:
1. Deprecated `additional_source_addresses`, then add new field called `extra_source_addresses`
2. Introduce a new field `additional_source_address_configs`, then match the additional source address and corresponding config by order.

@mattklein123 WDYT?

Reply:
I'd like the first one. 🤔

Reply:
> IMO, I'd like the first one. 🤔 Although the second one can avoid the deprecating, but it's a little weird that to match additional source address and corresponding config by order

Yea, the second one is pretty like a 'workaround', maybe not worth it.